### PR TITLE
settings: more guidance

### PR DESCRIPTION
**Pull request by knz**
Epic: CRDB-6671

As requested by @dt [here](https://github.com/cockroachdb/cockroach/pull/111579#discussion_r1342886114).

Release note: None

### CODE REVIEW

File: pkg/settings/setting.go [hunk 1]
```diff
@@ -190,12 +191,26 @@ const (
...
-	// Note that even though SystemVisible settings are meant to apply
-	// equally to all virtual clusters, we have a "break-the-glass" way
-	// to change the value observed by one or more virtual cluster,
-	// using ALTER VIRTUAL CLUSTER SET CLUSTER SETTING. This can be used
-	// during e.g. troubleshooting, to allow changing the observed value
-	// for a virtual cluster without a code change.
+	// One use cases is to enable optimizations in the KV client side of
+	// virtual cluster RPCs. For example, if the storage layer uses a
+	// setting to decide the size of a response, making the setting
+	// visible enables the client in virtual cluster ocde to
```
Comment (dt) [inline pkg/settings/setting.go:197]: s/ocde/code

**Review by dt — APPROVED** [review]

File: pkg/settings/setting.go (hunk 1 above)
Comment (knz) [reply pkg/settings/setting.go]: Fixed.

File: pkg/settings/setting.go [hunk 2]
```diff
@@ -190,12 +191,26 @@ const (
...
+	// rangefeeds and provide a better error message.
+	// (Without this extra logic, the error reported by KV/storage
+	// might be hard to read by end-users.)
+	//
+	// As a reminder, the setting values observed by a virtual cluster,
+	// both for its own application settings and those it observes for
+	// system-visible settings, can be overridden using ALTER VIRTUAL
+	// CLUSTER SET CLUSTER SETTING. This can be used during e.g.
+	// troubleshooting, to allow changing the observed value for a
+	// virtual cluster without a code change.
```
[the remark this answers is not in the document]
Comment (knz) [reply pkg/settings/setting.go:213]: Done.

**Comment by knz** [comment]
bors r=dt

**Pull request merged**